core/src/main/java/com/linecorp/armeria/server/streaming/JsonLines.java

```diff
@@ -0,0 +1,409 @@
+ * under the License.
+ */
+package com.linecorp.armeria.server.streaming;
+
+import static com.linecorp.armeria.internal.server.ResponseConversionUtil.streamingFrom;
+import static java.util.Objects.requireNonNull;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * A utility class which helps to create a <a href="https://jsonlines.org/">JavaScript Object
+ * Notation (JSON) Lines text</a> from a content {@link Publisher} or {@link Stream}.
+ *
+ * <p>A user simply creates a streaming {@link HttpResponse} which emits JSON Lines text, e.g.
+ * <pre>{@code
+ * ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.WRITE_ENUMS_USING_INDEX);
+ * Server server =
+ *     Server.builder()
+ *           // Emit JSON Lines Text with a default ObjectMapper.
+ *           .service("/seq1",
+ *                    (ctx, req) -> JsonLines.fromPublisher(Flux.just("foo", "bar")))
+ *           // Emit JSON Lines Text with the ObjectMapper
+ *           // configured to use SerializationFeature.WRITE_ENUMS_USING_INDEX.
+ *           .service("/seq2",
+ *                    (ctx, req) -> JsonLines.fromPublisher(Flux.just("foo", "bar"), mapper))
+ *           .build();
+ * }</pre>
+ */
+public final class JsonLines {
+    private static final Logger logger = LoggerFactory.getLogger(JsonLines.class);
+
+    /**
+     * A flag whether a warn log has been written if the given status code is not the {@link HttpStatus#OK}.
+     */
+    private static boolean warnedStatusCode;
+
+    /**
+     * A flag whether a warn log has been written if the given content type is not the
+     * {@link MediaType#JSON_LINES}.
+     */
+    private static boolean warnedContentType;
+
+    /**
+     * A line feed which indicates the end of a JSON Line.
+     */
+    private static final byte LINE_FEED = 0x0A;
+
+    /**
+     * A default {@link ObjectMapper} which converts the objects into JSON Line.
+     */
+    private static final ObjectMapper defaultMapper = new ObjectMapper()
+                                                          .disable(SerializationFeature.INDENT_OUTPUT);
```

Comment:
We newly introduced a default Jackson module.
```suggestion
    private static final ObjectMapper defaultMapper = 
            JacksonUtil.newDefaultObjectMapper().disable(SerializationFeature.INDENT_OUTPUT);
```

Reply:
Done